glean_parser/csharp.py

```diff
@@ -0,0 +1,166 @@
+# -*- coding: utf-8 -*-
+
+# This Source Code Form is subject to the terms of the Mozilla Public
+# License, v. 2.0. If a copy of the MPL was not distributed with this
+# file, You can obtain one at http://mozilla.org/MPL/2.0/.
+
+"""
+Outputter to generate C# code for metrics.
+"""
+
+import enum
+import json
+from pathlib import Path
+from typing import Any, Dict, List, Union  # noqa
+
+from . import metrics
+from . import pings
+from . import util
+
+
+def csharp_datatypes_filter(value: util.JSONType) -> str:
+    """
+    A Jinja2 filter that renders C# literals.
+
+    Based on Python's JSONEncoder, but overrides:
+      - lists to use `new string[] {}` (only strings)
+      - dicts to use mapOf (not currently supported)
+      - sets to use setOf (not currently supported)
+      - enums to use the like-named C# enum
+    """
+
+    class CSharpEncoder(json.JSONEncoder):
+        def iterencode(self, value):
+            if isinstance(value, list):
+                yield "new string[] {"
```

Comment:
Since this will break if passed a list not containing all strings, maybe we add an assertion here?

```
assert all(isinstance(x) for x in value)
```
?